src/common.hh

```diff
@@ -0,0 +1,62 @@
+#ifndef ACQUIRE_DRIVER_ZARR_COMMON_H
+#define ACQUIRE_DRIVER_ZARR_COMMON_H
+
+#include "prelude.h"
+
+#include "device/props/components.h"
+
+#include <filesystem>
+
+namespace fs = std::filesystem;
+
+namespace acquire::sink::zarr {
+struct ImageDims
+{
+    uint32_t cols;
+    uint32_t rows;
+
+    friend bool operator<=(const ImageDims& lhs, const ImageDims& rhs) noexcept
+    {
+        return (lhs.cols <= rhs.cols) && (lhs.rows <= rhs.rows);
+    }
+};
+
+namespace common {
+size_t
+bytes_of_type(const SampleType& type);
+
+size_t
+bytes_per_tile(const ImageDims& tile_shape, const SampleType& type);
+
+size_t
+frames_per_chunk(const ImageDims& tile_shape,
```

Comment:
Optional: I know this name isn't new, but I wonder if `tiles_per_chunk` would make these things and their calculations a little easier to reason about.

Reply:
It's a tossup for me. If Nathan seconds this, I'll change it.